working-group/test-suite.md

```diff
@@ -0,0 +1,47 @@
+# Test Suite
+
+These are the description of the test that we intend to build. This document is a scaffold until we have the tests implemented. At that point, the code itself will become the documentation. 
+
+## Requests
+
+Server expense serialization format to be declared in the content-type of the request. 
+
+Server should validate the content-type based in the request to ensure it's a content type (serialization format) that it supports 
+* Return `415 Unsupported Media Type` otherwise
+
+A GET method should be successful
+* Attempt a GET method request for introspection
+   query (serialized as URL encoded key-value pair)
+
+A POST method should be successful
+* Attempt a POST method for introspection
+   query (serialized in body)
+
+A GET method with query, operationName and variable should be successful
+
+A POST method with query, operationName and variable should be successful
+
+## Responses
+
+A 200 response should include a body. The body should include at least a `data` attribute or an `errors` attribute.
```

Comment:
```suggestion
A 200 response MUST include a body. If the body has `Content-Type` of `application/json`, `application/graphql` or `application/graphql+json`, the body MUST be a JSON object that conforms to the ["Response Format" as specified in the GraphQL Specification](https://spec.graphql.org/June2018/#sec-Response-Format).
```

Reply:
I think `application/graphql` is not meant to be used for responses, only for requests. It makes sense when sending a raw query string like so:

    curl -v -H "Content-Type: application/graphql" -d "{ hello }"  "localhost:3000/graphql"

It might be a valid response header, but not to represent the result of a GraphQL operation. I guess it could be used for something like a schema/query generator.

Reply:
I believe the content-type `application/graphql+json` can refer to a GraphQL JSON response document and the GraphQL request body. In other words, I think it should be a JSON document containing at least one of `data` or `errors` as top level attributes in a response. 

When sending a POST request, a client should send `application/graphql+json` in the general case because they are also sending a JSON document, this time including a `query`, `operationName` and `variables` as the top-level attributes. The GraphQL query itself, which is found in this object (in the case of a POST request) at the `query` parameter should conform to the GraphQL specification.

I'm not clear on whether `application/graphql` is a synonym for `application/graphql+json`. Hypothetically, `application/graphql` could be distinct and refer to a valid GraphQL document string. However, we then would not be able to use it in GraphQL over HTTP since at no time do we send _only a GraphQL query string_ to a server or receive _only a GraphQL query string_ from a server.